source/crud/tests/unified/updateMany-dots_and_dollars.yml

```diff
@@ -0,0 +1,138 @@
+description: "updateMany-dots_and_dollars"
+
+schemaVersion: "1.0"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name crud-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name coll0
+
+initialData: &initialData
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - { _id: 1, foo: {} }
+
+tests:
+  - description: "Updating document to set top-level dollar-prefixed key on 5.0+ server"
```

Comment:
Note that this file has no error-expecting tests, which was an original goal from the scope:

> Create CRUD spec tests for pre-5.0 servers that expect server-side errors when inserting, updating, or replacing documents with dollar-prefixed and dotted keys. These tests may complement the previous set of tests for 5.0+ servers, which do not expect an error.

While implementing update tests, I realized that update operations have historically had no dot/dollar validation since both characters are used throughout for field paths and operators. One could argue that the successful tests here also aren't doing much to test the driver, but I do think it's helpful to have some examples of the new server syntax handy (especially since this isn't yet documented). Also, successful tests for update _were_ an original goal from the scope.

Reply:
Tests as examples/documentation (as opposed to actual invocations of expected driver behavior) seem good to me.